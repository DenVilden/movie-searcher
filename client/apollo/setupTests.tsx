import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from '../pages/_app';

// eslint-disable-next-line jest/prefer-spy-on
HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) =>
  render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  );

export * from '@testing-library/react';
