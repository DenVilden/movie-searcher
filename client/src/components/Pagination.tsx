import React, { useState } from 'react';
import styled from 'styled-components';
import { Pagination as PaginationComponent } from '@material-ui/lab';

const StyledPagination = styled(PaginationComponent)`
  margin: auto;
`;

type Props = {
  refetch: ({ variables: { page } }: { variables: { page: number } }) => void;
  totalPages: number;
  currentPage: number;
  scrollToTop: React.MutableRefObject<HTMLElement>;
};

const Pagination = ({
  totalPages,
  refetch,
  currentPage,
  scrollToTop,
}: Props) => {
  const [page, setPage] = useState(currentPage);

  return (
    <StyledPagination
      count={totalPages}
      onChange={(_, value: number) => {
        setPage(value);
        refetch({ variables: { page: value } });
        scrollToTop &&
          scrollToTop.current.scrollIntoView({
            behavior: 'smooth',
            block: 'start',
          });
      }}
      page={page}
    />
  );
};

export default Pagination;
