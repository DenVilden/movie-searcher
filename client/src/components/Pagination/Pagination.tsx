import { useState } from "react";
import styled from "styled-components";
import { Pagination } from "@material-ui/lab";

const StyledPagination = styled(Pagination)`
  margin: auto;
`;

interface Props {
  refetch: (page: number) => void;
  totalPages: number;
  currentPage: number;
  scrollToTop: React.MutableRefObject<HTMLElement | null>;
}

const PaginationComponent = ({
  totalPages,
  refetch,
  currentPage,
  scrollToTop,
}: Props) => {
  const [page, setPage] = useState(currentPage);

  return (
    <StyledPagination
      count={totalPages}
      onChange={(_evt, value: number) => {
        setPage(value);
        refetch(value);
        scrollToTop.current?.scrollIntoView({
          behavior: "smooth",
          block: "start",
        });
      }}
      page={page}
    />
  );
};

export default PaginationComponent;
