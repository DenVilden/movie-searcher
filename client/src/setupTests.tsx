import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from './pages/_app';
import defaultResolvers from './graphql/resolvers';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
  cache?: any;
  resolvers?: any;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks, addTypeName, resolvers, cache }: RenderApolloOptions = {}
) =>
  render(
    <MockedProvider
      addTypename={addTypeName}
      cache={cache}
      mocks={mocks}
      resolvers={resolvers || defaultResolvers}
    >
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>
  );

export * from '@testing-library/react';
