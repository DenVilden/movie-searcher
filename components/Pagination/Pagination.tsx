import { useState } from 'react';
import styled from '@emotion/styled';
import { Pagination } from '@material-ui/core';
import { useRouter } from 'next/router';

const StyledPagination = styled(Pagination)`
  margin: auto;
  margin-bottom: 10px;

  ${(props) => props.theme.breakpoints.up('md')} {
    margin-bottom: 35px;
  }
`;

interface Props {
  totalPages: number;
  currentPage: number;
  path: string;
}

export default function PaginationComponent({
  totalPages,
  currentPage,
  path,
}: Props) {
  const [page, setPage] = useState(currentPage);
  const router = useRouter();

  return (
    <StyledPagination
      count={totalPages}
      page={page}
      onChange={(_evt, pageNumber: number) => {
        setPage(pageNumber);
        router.push(`/${path}/${pageNumber}`);
      }}
    />
  );
}
