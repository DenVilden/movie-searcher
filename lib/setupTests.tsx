/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/preact'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import '@testing-library/jest-dom/extend-expect'

const theme = createMuiTheme()

type RenderApolloOptions = {
  addTypeName?: boolean
  mocks?: MockedResponse[]
}

export function renderApollo(
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) {
  return render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  )
}

export { screen, fireEvent } from '@testing-library/preact'
