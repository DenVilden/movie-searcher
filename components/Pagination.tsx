import { useState } from 'react';
import styled from '@emotion/styled';
import Pagination from '@material-ui/core/Pagination';
import { useRouter } from 'next/router';

const StyledPagination = styled(Pagination)`
  margin: auto;
  margin-bottom: 10px;

  ${({ theme }) => theme.breakpoints.up('md')} {
    margin-bottom: 35px;
  }
`;

interface Props {
  currentPage: number;
  path: string;
  totalPages: number;
}

export default function PaginationComponent({
  totalPages,
  currentPage,
  path,
}: Props) {
  const [page, setPage] = useState(currentPage);
  const router = useRouter();

  return (
    <StyledPagination
      count={totalPages}
      onChange={(_evt, pageNumber: number) => {
        setPage(pageNumber);
        router.push(`/${path}/${pageNumber}`);
      }}
      page={page}
    />
  );
}
