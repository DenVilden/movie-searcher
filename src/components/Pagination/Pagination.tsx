import React from 'react';
import { StyledPagination } from './Pagination.styles';

const Pagination = ({
  totalPages,
  refetch,
  currentPage,
}: {
  refetch: ({ page }: { page: number }) => void;
  totalPages: number;
  currentPage: number;
}) => {
  const [page, setPage] = React.useState(currentPage);

  const handleChange = (_: React.SyntheticEvent, value: number) => {
    setPage(value);
    refetch({ page: value });
  };

  return (
    <StyledPagination count={totalPages} onChange={handleChange} page={page} />
  );
};

export default Pagination;
