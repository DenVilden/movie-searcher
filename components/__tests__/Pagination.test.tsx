import { renderApollo } from '../../lib/setupTests';
import Pagination from '../Pagination';

describe('pagination', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderApollo(
      <Pagination totalPages={10} currentPage={1} path="path" />,
    );

    const element = asFragment();

    expect(element).toMatchSnapshot();
  });
});
