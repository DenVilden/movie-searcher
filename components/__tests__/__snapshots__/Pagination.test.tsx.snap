// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root MuiPagination-text css-hhphb0"
  >
    <ul
      class="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-text Mui-disabled css-10d1a0h-MuiButtonBase-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-text Mui-selected css-10d1a0h-MuiButtonBase-root"
          tabindex="0"
          type="button"
        >
          1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-text css-10d1a0h-MuiButtonBase-root"
          tabindex="0"
          type="button"
        >
          2
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 3"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-text css-10d1a0h-MuiButtonBase-root"
          tabindex="0"
          type="button"
        >
          3
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 4"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-text css-10d1a0h-MuiButtonBase-root"
          tabindex="0"
          type="button"
        >
          4
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 5"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-text css-10d1a0h-MuiButtonBase-root"
          tabindex="0"
          type="button"
        >
          5
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <div
          class="MuiPaginationItem-root MuiPaginationItem-ellipsis"
        >
          â€¦
        </div>
      </li>
      <li>
        <button
          aria-label="Go to page 10"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-text css-10d1a0h-MuiButtonBase-root"
          tabindex="0"
          type="button"
        >
          10
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-text css-10d1a0h-MuiButtonBase-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
